File: ruby/f00.loop.rb
#!/usr/bin/env ruby
# collection loop from output of subshell
for item in `ls`.split.each do            # cycle through directory listing
   if File.directory? item                # test if path is directory
       puts "#{item} is a directory."
   else
       puts "#{item} is not a directory."
   end
end

File: php/f00.loop.php
#!/usr/bin/env php
<?php
// collection loop on results
foreach (preg_split("/\s+/", shell_exec('ls')) as $item) {
   if (is_dir($item)) {
       echo "$item is a directory\n";
   } else {
       echo "$item is not a directory\n";
   }
}
?>
